prow/cmd/jenkins-operator/main.go

```diff
@@ -60,12 +58,22 @@ type options struct {
 	caCertFile             string
 	csrfProtect            bool
 
-	githubEndpoint  flagutil.Strings
-	githubTokenFile string
-	dryRun          bool
+	dryRun     bool
+	kubernetes prowcmdutil.KubernetesOptions
+	github     prowcmdutil.GitHubOptions
 }
 
 func (o *options) Validate() error {
+	for _, group := range []cmdutil.OptionGroup{&o.kubernetes, &o.github} {
+		if err := group.Validate(o.dryRun); err != nil {
+			return err
+		}
+	}
+
+	if err := cmdutil.ValidateURIs("-jenkins-url", []string{o.jenkinsURL}); err != nil {
```

Comment:
Not sure we really need a helper for calling `url.Parse`

Reply:
> Not sure we really need a helper for calling `url.Parse`

I needed the helper for the GitHub and Kubernetes option groups, so I thought I'd recycle it here to get DRY error-message generation.

Reply:
nits:
- There is minimal value to DRYing the error message generation in my opinion. This is only used in a few places and is just a wrapper for `url.Parse`. It is easier to read the code without the extra layer of indirection and trying to guess what the first parameter does. If you feel strongly about this we can keep it, but it should at least be moved out of the `cmdutil` package. Validating URIs isn't specific to client initialization.
- We typically use `--flag-name` instead of `-flag-name` because the former is what is actually specified on the command line.

Reply:
> There is minimal value to DRYing the error message generation in my opinion...

That's two votes against, and it's your repo, so I'll drop it ;).

> We typically use --flag-name instead of -flag-name because the former is what is actually specified on the command line.

What do you mean "what is actually specified on the command line"?

```console
$ go build ./prow/cmd/hook
$ ./hook -help
Usage of ./hook:
  -cluster string
    	Path to kube.Cluster YAML file. If empty, uses the local cluster.
  -config-path string
...
```

That seems slanted towards single dashes, and having error messages match `-help` output seems more accessible.  It looks like Go's flag has supported double-dash forms [since the beginning][1], but that it's more canonical to use the single-dash forms.

[1]: https://github.com/golang/go/commit/ac0dd5ae525db0d057e94c03c3f506bc30afae31#diff-0ac2365b4b7c61c3b442a91aa219890aR354

Reply:
> > There is minimal value to DRYing the error message generation in my opinion...
>
> That's two votes against, and it's your repo, so I'll drop it ;).

Dropped with 6b11fd7 -> 804ad9c, which also rebases onto master.

Reply:
You are right, Go doesn't follow standard UNIX/GNU flag syntax: https://groups.google.com/forum/#!topic/golang-nuts/3myLL-6mA94
Single dashes seem fine then? Most everywhere else in Prow we use the double dash form though, so for consistency we may want to fix the other places or decide to commit to a non-Go flag syntax in a follow up.